pkg/x509/certificate.go

```diff
@@ -13,22 +13,28 @@ import (
 	"github.com/pkg/errors"
 )
 
-func CreateSelfSignedCert() (certPEM, keyPEM []byte, err error) {
+func CreateSelfSignedCert(certIPs []string, certDNSNames []string) (certPEM, keyPEM []byte, err error) {
 	// generate a new key-pair
 	rootKey, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		return
 	}
 
 	// generate cert template used to self-sign
-	certTmpl, err := certTemplate()
+	certTmpl, err := certTemplate(certDNSNames[0])
 	if err != nil {
 		return
 	}
 	certTmpl.IsCA = true
 	certTmpl.KeyUsage = x509.KeyUsageCertSign | x509.KeyUsageDigitalSignature
 	certTmpl.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth}
-	certTmpl.IPAddresses = []net.IP{net.ParseIP("127.0.0.1")}
+	certTmpl.DNSNames = certDNSNames
+
+	// parse ips
+	certTmpl.IPAddresses = []net.IP{}
```

Comment:
```suggestion
```
`append` will allocate the slice for you.

Reply:
Nuked